### ISSUE TITLE
Finger uses the string module

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5321]
|[<img alt="pitrou's avatar" src="https://avatars.githubusercontent.com/u/1721820?s=50" width="50" height="50">](https://github.com/pitrou)| @pitrou reported|
|-|-|
|Trac ID|trac#5321|
|Type|enhancement|
|Created|2011-10-19 13:10:46Z|
|Branch|https://github.com/twisted/twisted/tree/no-string-module-5321|

Fix the finger protocol to use methods of str objects, not functions in the string module. Patch attached.

Attachments:

* [finger_string_module.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/c17/c17ece1ef4818595d96d145b0b62751779d67f63/b7ffa61668984357a9db819d77518d591877606c.patch) (841 bytes) - added by antoine on 2011-10-19 13:10:53Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__5321 5321
type__enhancement enhancement
reporter__antoine antoine
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_no_string_module_5321 branches/no-string-module-5321
branch_author__teratorn teratorn
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1319029846000000 1319029846000000
changetime__1319338666000000 1319338666000000
version__None None
owner__teratorn teratorn
cc__antoine
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)|@teratorn set owner to @teratorn|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_4"></a>|@teratorn commented|
|-|-|

(In [32965]) Branching to 'no-string-module-5321'

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_5"></a>|@teratorn commented|
|-|-|

Looks good, thanks.

**Comment by twisted-trac** [comment]
|[<img alt="teratorn's avatar" src="https://avatars.githubusercontent.com/u/154069?s=50" width="50" height="50">](https://github.com/teratorn)<a name="note_6"></a>|@teratorn set status to `closed`|
|-|-|

(In [32968]) Merge no-string-module-5321: don't use string module functions as they are removed in Python 3
Author: Antoine Pitrou
Reviewer: teratorn
Fixes: [#5321](https://github.com/twisted/twisted/issues/5321)
